mypy/binder.py

```diff
@@ -103,19 +109,21 @@ def _get(self, key: Key, index: int=-1) -> Type:
                 return self.frames[i][key]
         return None
 
-    def push(self, node: Node, typ: Type) -> None:
-        if not node.literal:
+    def put_if_bindable(self, expr: Expression, typ: Type) -> None:
```

Comment:
I like the change to `put`, but `_if_bindable` doesn't add any information (there's no contrasting `put_even_if_not_bindable`) and is pretty long.

Basically, `put` is just the way for the rest of the type checker to pass information into the binder, and I think of it as the binder's business whether it bothers to record a piece of information it is given or not.